packages/gestalt/src/Tabs.js

```diff
@@ -4,48 +4,103 @@ import PropTypes from 'prop-types';
 import Box from './Box.js';
 import Flex from './Flex.js';
 import Link from './Link.js';
+import TapArea from './TapArea.js';
 import Text from './Text.js';
 import { type AbstractEventHandler } from './AbstractEventHandler.js';
 
 type OnChangeHandler = AbstractEventHandler<
-  SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  | SyntheticMouseEvent<HTMLAnchorElement>
+  | SyntheticKeyboardEvent<HTMLAnchorElement>
+  | SyntheticMouseEvent<HTMLDivElement>
+  | SyntheticKeyboardEvent<HTMLDivElement>,
   {| +activeTabIndex: number |},
 >;
 
-function Circle() {
+function Dot() {
   return (
     <Box
       color="red"
       dangerouslySetInlineStyle={{ __style: { marginTop: '1px' } }}
       height={6}
-      marginStart={2}
       rounding="circle"
       width={6}
     />
   );
 }
 
+const UNDERLINE_HEIGHT = 3;
+
+function Underline() {
+  return (
+    <Box
+      color="darkGray"
+      dangerouslySetInlineStyle={{
+        __style: {
+          borderRadius: 1.5,
+        },
+      }}
+      height={UNDERLINE_HEIGHT}
+      width="100%"
+    />
+  );
+}
+
+const COUNT_HEIGHT_PX = 16;
+
+function Count({ count }: {| count: number |}) {
+  const displayCount = count < 100 ? `${count}` : '99+';
```

Comment:
In addition to truncating values over 100, we need to know the number of digits we're displaying, so stringifying here